Update drug index resolution of MoA and Indications

JarrodBaker opened the pull request:
Updated GraphQL resolving logic to meet specification in ticket opentargets/platform#1308

- Indications are not inherited in either direction
- MoA are treated as a family of molecules, so parent, child and sibling all have the same MoAs and Linked Targets
- Add and expose parentId to model
- Expose childIds

mirandaio commented on README.md:
```diff
@@ -4,7 +4,7 @@ Experimental GraphQL API to serve `platform-app`.
 ## How to use
 
-Clone the repo, `yarn install`, `yarn start`. Browse to http://localhost:4000/graphql.
+Clone the repo, `yarn install`, `yarn build` and `yarn start`. Browse to http://localhost:4000/graphql.
```
browse to http://localhost:8080/graphql now, no?

JarrodBaker committed:
Adding MechanismsOfAction as linked entity on drug using own elasticsearch index.

JarrodBaker committed:
Adding Indication as linked entity on drug using own elasticsearch index.

JarrodBaker committed:
Updating Indication so indications that appear on a parent (via children) are visible on all children.

JarrodBaker committed:
Updating Linked Targets so parents and children all show the same output.

JarrodBaker committed:
Update mechanism of action GraphQL propagation

All molecules in the family should have the same mechanisms of action.
Updated to ensure that children can inherit from parents and siblings.

Relates to: opentargets#1308

JarrodBaker committed:
Revert Indication hierarchical accumulation from c3bb80c05c1bb099218484e73a840815a646d77d

JarrodBaker committed:
Update logging and test feedback

JarrodBaker committed:
Optimise imports on Objects.scala

d0choa commented:
I'm confused with the first bullet point: "Indications present on children are also present on parent".

What we would like to have for indications is that the parent indications stay with the parent and the child indications stay with the child. It's the same implementation you had prior to this PR and respects what ChEMBL does on their own website.

Does it make sense?

JarrodBaker commented:
> I'm confused with the first bullet point: "Indications present on children are also present on parent".
> ...
> Does it make sense?

Yes, I'll amend. Thanks!

JarrodBaker committed:
Update Indication to remove parent inheriting from children.

JarrodBaker committed:
Remove unused Fetcher for Mechanism of Action
